hazelcast/src/main/java/com/hazelcast/client/impl/protocol/MessageTaskFactoryProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol;
+
+/***
+ * Provider for client's Message task factories;
```

Comment:
Can you add some comment how this interface fits into the picture?

Should this class not be in the com.hazelcast.client package since AFAIK it is public API. If outside developer wants to add support for a new data-structure, he needs to make a implementation of the MessageTaskFactoryProvider interface.